src/codemodder/context.py

```diff
@@ -80,16 +89,39 @@ def __init__(
         self.tool_result_files_map = tool_result_files_map or {}
         self.llm_client = self._setup_llm_client()
 
-    def _setup_llm_client(self) -> Client | None:
-        if not Client:
-            logger.debug("OpenAI API client not available")
+    def _setup_llm_client(self) -> OpenAI | None:
+        if not AzureOpenAI:
+            logger.info("Azure OpenAI API client not available")
+            return None
+
+        azure_openapi_key = os.getenv("CODEMODDER_AZURE_OPENAI_API_KEY")
+        azure_openapi_endpoint = os.getenv("CODEMODDER_AZURE_OPENAI_ENDPOINT")
+        if bool(azure_openapi_key) ^ bool(azure_openapi_endpoint):
```

Comment:
could use walrus here for 2 lines above

Reply:
That's true although I think in this case it might make it a little harder to read 😅